### PR TITLE
Support zstd (de)compression

### PR DESCRIPTION
**Pull request by davidpdrsn**
Fixes https://github.com/tower-rs/tower-http/issues/319

### CODE REVIEW

File: tower-http/src/compression_utils.rs [hunk 1]
```diff
@@ -20,20 +20,29 @@ pub(crate) struct AcceptEncoding {
...
+            (false, false, true, false) => "br",
+            (true, true, true, true) => "gzip,deflate,br,zstd",
+            (true, true, false, true) => "gzip,deflate,zstd",
+            (true, false, true, true) => "gzip,br,zstd",
+            (true, false, false, true) => "gzip,zstd",
+            (false, true, true, true) => "deflate,br,zstd",
+            (false, true, false, true) => "deflate,zstd",
+            (false, false, true, true) => "br,zstd",
+            (false, false, false, true) => "zstd",
+            (false, false, false, false) => return None,
```
Comment (davidpdrsn) [inline tower-http/src/compression_utils.rs:45]: Not really sure what the priorities should be here.

File: tower-http/src/content_encoding.rs [hunk 2]
```diff
@@ -26,6 +29,8 @@ impl Encoding {
             Encoding::Deflate => "deflate",
             #[cfg(any(feature = "fs", feature = "compression-br"))]
             Encoding::Brotli => "br",
+            #[cfg(any(feature = "fs", feature = "compression-zstd"))]
+            Encoding::Zstd => "br",
```
Comment (Nehliin) [inline tower-http/src/content_encoding.rs:33]: Copy paste error :)

File: tower-http/src/compression_utils.rs (hunk 1 above)
Comment (jplatte) [reply tower-http/src/compression_utils.rs:45]: zstd is great, zstd first x)

File: tower-http/src/services/fs/serve_dir/mod.rs [hunk 3]
```diff
@@ -180,6 +180,23 @@ impl<F> ServeDir<F> {
         self
     }
 
+    /// Informs the service that it should also look for a precompressed zstd
+    /// version of _any_ file in the directory.
+    ///
+    /// Assuming the `dir` directory is being served and `dir/foo.txt` is requested,
+    /// a client with an `Accept-Encoding` header that allows the zstd encoding
+    /// will receive the file `dir/foo.txt.zz` instead of `dir/foo.txt`.
```
Comment (Nehliin) [inline tower-http/src/services/fs/serve_dir/mod.rs:188]: .zstd here instead

**Review by Nehliin — CHANGES_REQUESTED** [review]
Nice, just found some small copy paste errors

File: tower-http/src/compression_utils.rs (hunk 1 above)
Comment (Nehliin) [reply tower-http/src/compression_utils.rs:45]: Br is supposedly pretty good as well, and I guess more available than zstd?

File: tower-http/src/services/fs/serve_dir/mod.rs (hunk 3 above)
Comment (zamazan4ik) [reply tower-http/src/services/fs/serve_dir/mod.rs:188]: If we are talking about files, I guess `.zst` is the right choice: https://www.iana.org/assignments/media-types/application/zstd

File: tower-http/src/compression_utils.rs (hunk 1 above)
Comment (jplatte) [reply tower-http/src/compression_utils.rs:45]: I wasn't really serious, by the way. But I agree with @Nehliin, if we want to sort by quality then `br` should also be near the start.

**Review by Nehliin — APPROVED** [review]
LGTM

**Pull request merged**